builtin/credential/aws/path_role.go

```diff
@@ -20,6 +20,12 @@ func pathRole(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "Name of the role.",
 			},
+			"allowed_auth_types": {
```

Comment:
Usually in this type of situation, we put in logic so that existing roles will maintain the current status quo and new roles will get the new behavior.

An easy way to do this would be, on role read of an existing role, to check if `allowed_auth_types` is set at all (since it's a new variable), and if so, set it to `ec2`, but set the default for new roles to `ec2,iam`. This will work since any new roles won't successfully load an existing entry.